src/safir/logging.py

```diff
@@ -0,0 +1,172 @@
+"""Utilities for configuring structlog-based logging.
+"""
+
+__all__ = ["configure_logging", "response_logger", "get_response_logger"]
+
+import logging
+import sys
+from contextvars import ContextVar
+from typing import TYPE_CHECKING
+
+import structlog
+
+if TYPE_CHECKING:
+    from structlog._config import BoundLoggerLazyProxy
+
+
+response_logger: ContextVar["BoundLoggerLazyProxy"] = ContextVar(
+    "response_logger"
+)
+"""A context-local structlog logger.
+
+This logger is set by the `safir.middleware.bind_logger` middleware.
+
+See also
+--------
+get_response_logger
+
+Examples
+--------
+Usage:
+
+>>> logger = response_logger.get()
+>>> logger.info(key='value')
+"""
+
+
+def get_response_logger() -> "BoundLoggerLazyProxy":
+    """Get the context-local structlog logger with bound request context.
+
+    This logger is set by `safir.middleware.bind_logger`.
+
+    Returns
+    -------
+    logger : BoundLoggerLazyProxy
+        A context-local structlog logger.
+
+    See also
+    --------
+    response_logger
+
+    Examples
+    --------
+    Usage:
+
+    .. code-block:: python
+
+       from safir.logging import get_response_logger
+       logger = get_response_logger()
+       logger.info('Some message', somekey='somevalue')
```

Comment:
Nit: You're using single quotes in the Python examples in the documentation, but black will fix all quoting to use double quotes. Might be worth using double quotes in the documentation examples as well for consistency.